src/views/studio/CreateEditChannelView/CreateEditChannelView.tsx

```diff
@@ -46,220 +53,240 @@ type CreateEditChannelViewProps = {
+    const coverAssetUrl = createUrlFromAsset(coverPhotoAvailability, coverPhotoUrl, coverPhotoDataObject)
+
+    const foundLanguage = languages.find(({ value }) => value === language?.iso)
+
     reset({
-      ...getValues(),
-      avatar: { blob: null, url: avatarPhotoUrl },
-      cover: { blob: null, url: coverPhotoUrl },
-      channelName: title || '',
+      avatar: { blob: null, url: avatarAssetUrl },
+      cover: { blob: null, url: coverAssetUrl },
+      title: title || '',
       description: description || '',
-      isPublic,
-      selectedLanguage: foundLanguage || languages[0],
+      isPublic: isPublic ?? false,
+      language: foundLanguage?.value || languages[0].value,
     })
-    if (avatarPhotoUrl) {
-      setAvatarImageUrl(avatarPhotoUrl)
-    }
-    if (coverPhotoUrl) {
-      setCoverImageUrl(coverPhotoUrl)
-    }
-  }, [channel, getValues, loading, newChannel, reset])
+  }, [channel, loading, newChannel, reset])
 
-  const handleSubmit = useHandleSubmit((data) => {
-    console.log(data)
+  const handleSubmit = createSubmitHandler(async (data) => {
+    console.log({ dirtyFields })
+    console.log({ data })
+    setTransactionStatus(ExtrinsicStatus.Unsigned)
   })
-  const handleAvatarConfirm = (blob: Blob, url: string) => {
-    setValue('avatar', { url, blob })
-    setAvatarImageUrl(url)
-    setActionBarActive(true)
-  }
-  const handleCoverConfirm = (blob: Blob, url: string) => {
-    setValue('cover', { url, blob })
-    setCoverImageUrl(url)
-    setActionBarActive(true)
-  }
-  const handleCoverRemove = () => {
-    setValue('cover', null)
-    setCoverImageUrl(null)
-    setActionBarActive(true)
+
+  const handleTransactionCancel = () => {
+    setTransactionStatus(null)
   }
 
   if (error) {
     throw error
   }
 
   return (
-    <form onSubmit={handleSubmit}>
-      <Header>
-        <ChannelCover
-          coverPhotoUrl={coverImageUrl}
-          onCoverEditClick={(e) => {
-            e.preventDefault()
-            coverDialogRef.current?.open()
-          }}
-          onCoverRemoveClick={handleCoverRemove}
-          editable
-        />
-
-        <StyledTitleSection className={transitions.names.slide}>
-          <StyledAvatar
-            imageUrl={avatarImageUrl}
-            size="fill"
-            onEditClick={(e) => {
-              e.preventDefault()
-              avatarDialogRef.current?.open()
-            }}
-            editable
+    <>
+      <TransactionDialog status={transactionStatus} onCancel={handleTransactionCancel} />
+      <form onSubmit={handleSubmit}>
+        <Header>
+          <Controller
+            name="cover"
+            control={control}
+            render={({ value, onChange }) => (
+              <>
+                <ChannelCover
+                  coverPhotoUrl={loading ? null : value.url}
+                  onCoverEditClick={(e) => {
+                    e.preventDefault()
```

Comment:
maybe we don't need that anymore since you've changed button's type?

Reply:
yes, those weren't needed, removed